tracker_test.go

```diff
@@ -79,24 +99,24 @@ func TestTrackerListItemsNotNil(t *testing.T) {
 			t.Error("Returned Tracker found nil")
 		}
 	}
-	test.DeleteTrackerOK(t, nil, nil, &controller, item1.ID)
-	test.DeleteTrackerOK(t, nil, nil, &controller, item2.ID)
+	test.DeleteTrackerOK(t, nil, nil, &controller, *item1.Data.ID)
+	test.DeleteTrackerOK(t, nil, nil, &controller, *item2.Data.ID)
 }
 
 // This test ensures that ID returned by Show is valid.
 // refer : https://github.com/almighty/almighty-core/issues/189
 func TestCreateTrackerValidId(t *testing.T) {
 	resource.Require(t, resource.Database)
 	controller := TrackerController{Controller: nil, db: gormapplication.NewGormDB(DB), scheduler: RwiScheduler}
-	payload := app.CreateTrackerAlternatePayload{
+	payload := getTrackerPayload(trackerAttr{
 		URL:  "http://issues.jboss.com",
 		Type: "jira",
-	}
+	})
 	_, tracker := test.CreateTrackerCreated(t, nil, nil, &controller, &payload)
-
-	_, created := test.ShowTrackerOK(t, nil, nil, &controller, tracker.ID)
-	if created != nil && created.ID != tracker.ID {
-		t.Error("Failed because fetched Tracker not same as requested. Found: ", tracker.ID, " Expected, ", created.ID)
+	trackerID := *tracker.Data.ID
+	_, created := test.ShowTrackerOK(t, nil, nil, &controller, trackerID)
+	if created != nil && created.ID != trackerID {
+		t.Error("Failed because fetched Tracker not same as requested. Found: ", trackerID, " Expected, ", created.ID)
 	}
-	test.DeleteTrackerOK(t, nil, nil, &controller, tracker.ID)
+	test.DeleteTrackerOK(t, nil, nil, &controller, trackerID)
```

Comment:
Use `gormsupport.DeleteCreatedEntities` as a deferred call. There are many examples in other files.